hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/IntoClause.java

```diff
@@ -160,10 +160,8 @@ private void visitPropertySpecNodes(AST propertyNode, List types) {
 			if ( persister.getIdentifierType() instanceof CompositeType ) {
 				if ( componentIds == null ) {
 					String[] propertyNames = ( (CompositeType) persister.getIdentifierType() ).getPropertyNames();
-					componentIds = new HashSet();
-					for ( String propertyName : propertyNames ) {
-						componentIds.add( propertyName );
-					}
+					componentIds = new HashSet(propertyNames.length);
+					Collections.addAll(componentIds,propertyNames);
```

Comment:
I think the above two statements could be combined with
`
componentIds = new HashSet(propertyNames);
`